### DISCUSSION TITLE
Failed to load the process imported by 'CefSharp. Core. Runtime. dll'

### DISCUSSION THREAD

**Discussion by 9death** [cefsharp/CefSharp#4409] [General]
I integrated cefsharp, packaged it into dll, placed it in the folder specified by revit, opened revit, and prompted this error

**Comment by amaitland** [comment]
Insufficient information, please answer the questions at https://github.com/cefsharp/CefSharp/wiki/Questions-&-Support#when-asking-a-question-please-include-the-following

**Reply by AmenJlili** [reply]
I'm getting a similar exception.
``` {"A procedure imported by 'CefSharp.Core.Runtime.dll' could not be loaded.":"CefSharp.Core.Runtime.dll"} ```

What's interesting is that this doesn't happen in WPF app. It seems that System.Windows.Application can load the necessary dependencies correctly. The exception is very cryptic. 

I have all the VC++ redist installed on my sys.

**Reply by 9death** [reply]
The Cef version I use is 99.2, and the Revit version I use is 2018. I have no problem on my computer, but the above error is reported when I deploy it to someone else's computer. VC++redist has been installed on someone else's computer, but the error is still reported

**Reply by AmenJlili** [reply]
Are you attempting to load any dependencies using Assembly.LoadFrom ? 

Would to see some code. 


Also what VC++ are you using ?

**Reply by 9death** [reply]
![1](https://user-images.githubusercontent.com/22311212/222323913-8a93cea9-c9ac-404e-aef2-c573f76f87bb.png)
![2](https://user-images.githubusercontent.com/22311212/222323919-8ce7f39d-1e8c-44f1-a801-60655338884b.png)

**Reply by amaitland** [reply]
Only a single version of `CEF` can be loaded per **process**.

> The Cef version I use is 99.2, and the Revit version I use is 2018. I have no problem on my computer, but the above error is reported when I deploy it to someone else's computer. VC++redist has been installed on someone else's computer, but the error is still reported

From the little I know about `Revit` older versions shipped with a version of `CefSharp`. It's likely your user has loaded an older version of `CEF` in their process causing an exception when you attempt to load a newer version.

You will need to:
- Use the exact same version of `CefSharp` as `Revit` ships with.
- Run the browser in a different process (there's a demo at https://github.com/cefsharp/CefSharp.OutOfProcess)

For background https://forums.autodesk.com/t5/revit-api-forum/revit-2019-1-add-in-and-cefsharp-library/td-p/8205740

**Reply by AmenJlili** [reply]
Turns out the host app already has a new version of CEF. I was lucky enough to see that in the docs.

**Comment by AmenJlili** [comment]
I'm getting error: 

![image](https://user-images.githubusercontent.com/16106587/222307164-58773952-0f6f-4180-bb7e-ccf1ae566ea4.png)
